### DISCUSSION TITLE
Cookies attributes access

### DISCUSSION THREAD

**Discussion by Sim4n6** [mitmproxy/mitmproxy#5995] [Q&A] [answered]
Please, how to check the attributes of a cookie in a request.

I mean, I would like to do something like:

```python
for cookie in flow.request.cookies:
        print(cookie.key, cookie.value, cookie['expires'], cookie['domain'], cookie['path'], cookie['samesite'])
```

**Comment by Prinzhorn** [comment, marked as answer]
That doesn't make sense, all these attributes/properties don't exist on the request cookies. They are pure key/value pairs.
You can do that with the set-cookie headers using `flow.response.cookies`, see docs at https://docs.mitmproxy.org/stable/api/mitmproxy/http.html#Response.cookies

**Reply by Sim4n6** [reply]
Well I was looking for some cookie 🍪 jar with attributes but thank you I will check that.

**Reply by Prinzhorn** [reply]
I see, but mitmproxy does not keep redundant cookies in a jar, that's the browser's job. You should be able to create a makeshift jar and store all the attributes from the `response` header and later access them from the `request` hook. But that depends entirely on what you need to achieve, as things like expiration etc. will make this more complex than just a `host -> cookies` dict.

**Reply by Sim4n6** [reply]
Thank you for the reply, you are really very helping me out.

By the way, I want to replay a copied & modified request while filtering out cookies with `SameSite` set to `Lax` (something similar). 

Please, I come up with the following way to do that, `extract` details from `request(flow)` and `response(flow)`. Then somehow initiate a `requests.get()` with the collected details including cookies. I successfully did that using mitmproxy addon scripting. But it was somehow not well-written and I'm in need of more control. That is why I'm going with `requests` module ... 

what do you think, please?

**Reply by Prinzhorn** [reply]
Please open a separate discussion, as this sounds like a different problem.

**Reply by Sim4n6** [reply]
No need for that, thank you very much.
